File: proga/1_semestr/labs/lab3/src/main/java/dev/lysmux/lab3/util/Direction.java
package dev.lysmux.lab3.util;

public class Direction {
    public enum Preposition {
        IN("в"),
        ON("над"),
        UNDER("под"),
        IN_FRONT_OF("перед"),
        BEHIND("позади"),
        BETWEEN("между");

        private final String value;

        Preposition(String value) {
            this.value = value;
        }

        public String getValue() {
            return value;
        }
    }

    protected final String content;
    protected final Preposition preposition;

    public Direction(Preposition preposition, String content) {
        this.content = content;
        this.preposition = preposition;
    }

    @Override
    public String toString() {
        return preposition.getValue() + " " + content;
    }
}

File: proga/1_semestr/labs/lab3/src/main/kotlin/dev/lysmux/lab3/util/Direction.kt
package dev.lysmux.lab3.util

class Direction(private val preposition: Preposition, private val content: String) {
    enum class Preposition(val value: String) {
        IN("в"),
        ON("над"),
        UNDER("под"),
        IN_FRONT_OF("перед"),
        BEHIND("позади"),
        BETWEEN("между");
    }

    override fun toString() = "${preposition.value} $content"
}